### PR TITLE
Should not assume to exist outer commands.

### PR DESCRIPTION
**Pull request by unak**
They are not exist on Windows.

### CODE REVIEW

**Comment by eregon** [comment]
Agreed, are you looking at Windows failures?
I tried to set up AppVeyor recently to see how this project would run on Windows.
https://ci.appveyor.com/project/eregon/spec/build/6.
~500 failures on ~25000 tests, so not so bad but there is plenty of specs to filter or fix.

File: core/io/close_spec.rb [hunk 1]
```diff
@@ -50,7 +50,7 @@
 describe "IO#close on an IO.popen stream" do
 
   it "clears #pid" do
-    io = IO.popen 'yes', 'r'
+    io = IO.popen "#{RUBY_EXE} -e 'r = loop{puts \"y\"; 0} rescue 1; exit r'", 'r'
```
Comment (eregon) [inline core/io/close_spec.rb]: One way to avoid the escaping would be to put this code in a fixture file.

**Comment by unak** [comment]
> Agreed, are you looking at Windows failures?

Yes.  But I'm only picking out the most interesting (or critical) failures :-P

**Comment by vais** [comment]
@unak as a user of Ruby on Windows, I just want to say big thanks to you for looking into this! :beers:

**Pull request merged**